homework/App.js

```diff
@@ -7,58 +7,50 @@ class App {
     this.initialize(url);
   }
 
-  /**
-   * Initialization
-   * @param {string} url The GitHub URL for obtaining the organization's repositories.
-   */
   async initialize(url) {
-    // Add code here to initialize your app
-    // 1. Create the fixed HTML elements of your page
-    // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
-
     const root = document.getElementById('root');
-
-    Util.createAndAppend('h1', root, { text: 'It works!' }); // TODO: replace with your own code
-
+    const imgContainer = Util.createAndAppend('div', root, { class: 'logo' });
+    Util.createAndAppend('img', imgContainer, {
+      src: './hyf.png',
+      id: 'hyf-logo',
+      alt: 'logo image',
+    });
+    const header = Util.createAndAppend('header', root, { class: 'header' });
+    Util.createAndAppend('p', header, { html: 'HYF Repositories' });
+    const select = Util.createAndAppend('select', header, {
+      class: 'selectBox',
+      'aria-label': 'HYF Repositories',
+    });
+    Util.createAndAppend('div', root, { id: 'container' });
     try {
-      const repos = await Util.fetchJSON(url);
-      this.repos = repos.map(repo => new Repository(repo));
-      // TODO: add your own code here
+      const fetch = await Util.fetchJSON(url);
+      select.addEventListener('change', () => this.fetchContributorsAndRender(select.value));
+      Util.createAndAppend('div', root, { id: 'container' });
+      this.repositories = fetch
+        .sort((a, b) => a.name.localeCompare(b.name))
+        .map(repo => new Repository(repo));
+      this.repositories.forEach((a, b) => {
```

Comment:
The names `a` and `b` are meaningless. Your code would be easier to understand if you would use more descriptive names, e.g.:

```js
this.repositories.forEach((repository, index) => {
```